src/index.js

```diff
@@ -12,7 +12,7 @@ export type EqualityFn = (newArgs: mixed[], lastArgs: mixed[]) => boolean;
 export default function<ResultFn: (...any[]) => mixed>(
   resultFn: ResultFn,
   isEqual?: EqualityFn = areInputsEqual,
-): ResultFn {
+): ResultFn & { reset: () => void } {
```

Comment:
Thoughts @TrySound? @theKashey?

Reply:
```js
type ResultFn<T> = {|
  [[call]]: T,
  reset: () => void
|}
```

Reply:
Isn't `[[call]]` a very new syntax? I'm not sure about how it will be supported in projects that uses old Flow versions.
At work we still uses `0.85` because reasons

Reply:
Then ok, if it works.

Reply:
Then you will break TS. Any reasons why it should be inlined?
What it export `reset` as a separate command for the "consumers", while it could just call `fn.reset` internally. 

```js
import memoizeOne, {resetOne} from "memoize-one"

const fn = memoizeOne(realFunction);
resetOne(fn)
```

Reply:
Mixing default and named exports is a bad idea. Projects without esm will be broken.

Reply:
You mean that rollup magic, which already making cjs and esm exports not consistent would be broken? Yep, it would.
Probably this small, but crucial detail should be documented somewhere - better just before `export default`

Reply:
Nope, I mean webpack magic will be broken. With mixed exports rollup will treat modules in your favourite way.

Here's the issue when commonjs consumes esm with mixed exports
https://github.com/mui-org/material-ui/issues/11550

Reply:
There is - cjs bundle has `module.exports = memoize`, not `exports.default = memoize` how it _"should be"_. That was a problem with TS/Jest integration a while ago.

Reply:
It's simplification which TS can't deal with.

Reply:
@theKashey Can you explain why it breaks TS?
In `_.debounce` it's being done [that way](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/lodash/common/function.d.ts#L672).
And if you want to be 100% TS correct you need to break compatibility with current TS `memoize-one` definitions, and do something such as this:
```ts
declare function memoizeOne<A extends any[], R>(resultFn: (...args: A) => R, isEqual: NotImportantNow): {
  (...args: A): R;
  reset(): void,
};
```

`<T extends (...args: any[]) => any>` generic isn't correct, because it breaks types for functions that have properties

Reply:
Not "break" TS. It's more about - while Flow type definitions are here, and would be autoupdated - `d.ts` is not a part of this repo, and would not.

Reply:
So there are a few options:
1. Creating TS definitions as part of this repository
2. After (if) merge, creating a PR for [`@types`](https://github.com/DefinitelyTyped/DefinitelyTyped)
3. If it's such a breaking change (which I personally think it's not), updating major version, if this feature get merged

Anyway, I don't think this feature should raise more than a minor version, because only in very extreme cases adding `reset` property on the memoized functions will result in unwanted side effects